src/java.base/unix/classes/sun/nio/fs/UnixFileSystem.java

```diff
@@ -913,6 +914,9 @@ void move(UnixPath source, UnixPath target, CopyOption... options)
             if (!flags.replaceExisting) {
                 throw new FileAlreadyExistsException(
                     target.getPathForExceptionMessage());
+            } else if (!sourceAttrs.isSymbolicLink() || flags.followLinks) {
+                // Ensure source can be read
+                provider.checkAccess(source, AccessMode.READ);
```

Comment:
I don't think this will work with a SM set as that will trigger a check that the SM will allow read before it checks if there is read access at the file system level (move is specified to check the SM for write access to both source and target).

Reply:
I modified it in f09b468031ad542ba11e491a8091ea96a12d311d only to verify that the source has write access at the file system level.

I think that checkRead(), however, should be called if the move is effected as a copy + delete instead of as a rename. In this case,  move() calls copyFile() which does not check with the SM that there is permission to read the source. In copy(), checkRead() is called before copyFile() is invoked.

Reply:
The spec for Files.move is that the SM is called to checkWrite on both source and target. I would need to dig into ancient history to page in the reasons why it's not checkDelete on the source and checkWrite on the target.  In any case, for the change here, it can't introduce a checkRead without doing a spec change. So checking it to use provider.checkAccess is right as that will just check file access.

Reply:
> So checking it to use provider.checkAccess is right as that will just check file access.

Here I think you must've intended `access()` and not `checkAccess()`.